src/pretalx/schedule/models/availability.py

```diff
@@ -35,3 +35,100 @@ def serialize(self):
             # make sure all-day availabilities are displayed properly in fullcalendar
             'allDay': (self.start.time() == zerotime and self.end.time() == zerotime)
         }
+
+    def __str__(self):
+        return f'<Availability: start={self.start} end={self.end}>'
+
+    def overlaps(self, other, strict):
+        """ test if two Availabilities overlap or are directly adjacent to eachother, if not in strict mode """
+        assert isinstance(other, Availability)
+        if strict:
+            return (self.start <= other.start < self.end) or \
+                   (self.start < other.end <= self.end) or \
+                   (other.start <= self.start < other.end) or \
+                   (other.start < self.end <= other.end)
+        else:
+            return (self.start <= other.start <= self.end) or \
+                   (self.start <= other.end <= self.end) or \
+                   (other.start <= self.start <= other.end) or \
+                   (other.start <= self.end <= other.end)
+
+    def merge_with(self, other):
+        """ return a new Availability, which spans the whole range of this one and the given one """
+        assert other.overlaps(self, False)
```

Comment:
Please do not `assert` here – an `if … raise` is not only more explicit, it can also not be disabled (technically) by a Python interpreter setting.